### PR TITLE
Migrate Debian based builds to bookworm

### PR DESCRIPTION
**Pull request by tobim**
Since bookworm is now in hard freeze this upgrade should be safe. This will allow us to bump the minimum version of GCC to 12 should the need arise.

### CODE REVIEW

**Comment by tobim** [comment]
CI should run now after the Apache JFrog issues are resolved.
The only remaining question is whether we should bump the minimum requirement for GCC already, since 10 will not be tested anymore once this PR is merged.

File: .github/workflows/vast.yaml [hunk 1]
```diff
@@ -775,11 +775,11 @@ jobs:
       matrix:
         vast:
           - os: ubuntu-20.04
-            container: python:3.10-slim-bullseye
+            container: debian:bookworm-slim
             name: Debian
             compiler: GCC
-            cc: gcc-10
-            cxx: g++-10
+            cc: gcc-12
```
Comment (dominiklohmann) [inline .github/workflows/vast.yaml:781]: I think we should update the table in the docs in build.md already to gcc12+ and clang17+, as those are now the only versions we actively test with.

**Review by dominiklohmann — APPROVED** [review]
<3

File: Dockerfile [hunk 2]
```diff
@@ -95,18 +95,18 @@ RUN apt-get update && \
...
       ca-certificates \
       gnupg2 \
-      libasan5 \
+      libasan6 \
       libc++1 \
       libc++abi1 \
-      libflatbuffers1 \
-      libfmt7 \
+      libflatbuffers2 \
+      libfmt9 \
```
[the remark this answers is not in the document]
Comment (tobim) [reply Dockerfile:102]: It is still at 8 in nixpkgs because nobody bothered to bump it for spdlog yet.

File: .github/workflows/vast.yaml (hunk 1 above)
Comment (patszt) [reply .github/workflows/vast.yaml:781]: Nice, i am actually struggling with my code not working on gcc 10 due to a bug. I am trying to update to gcc 11 myself. Can't wait for it to get merged :D

File: cmake/VASTRegisterPlugin.cmake [hunk 3]
```diff
@@ -160,7 +160,15 @@ function (VASTCompileFlatBuffers)
   add_library(${FBS_TARGET} INTERFACE)
 
   # Link our target against FlatBuffers.
-  find_package(Flatbuffers REQUIRED CONFIG)
+  find_package(Flatbuffers CONFIG)
+  # Ugly workaround for Flatbuffers 2.0.8, which briefly changed the
+  # project name to "FlatBuffers".
```
Comment (dominiklohmann) [inline cmake/VASTRegisterPlugin.cmake:165]: The same workaround needs to be appended to `VAST_FIND_DEPENDENCY_LIST` right below.

**Pull request merged**